src/components/HomePage.js

```diff
@@ -0,0 +1,62 @@
+import React, { useEffect } from 'react';
+import { useDispatch, useSelector } from 'react-redux';
+import { Link } from 'react-router-dom';
+import { FaArrowRightLong } from 'react-icons/fa6';
+import { setCountryName, fetchCountryData } from '../redux/actions';
+
+const HomePage = () => {
+  const dispatch = useDispatch();
+  const countryName = useSelector((state) => state.countryName);
+  const countryData = useSelector((state) => state.countryData);
+
+  useEffect(() => {
+    dispatch(fetchCountryData(countryName));
+  }, [dispatch, countryName]);
+
+  return (
+    <div>
+      <div className="bg-gradient-to-b from-sky-500 to-sky-700 p-4 grid grid-cols-3 md:grid-cols-6 gap-2 md:gap-6 text-white">
+        <h1>Country Metrics</h1>
+        <input
+          type="text"
+          className="p-1 border-2 border-white ease-in rounded-md text-sky-700"
+          placeholder="Enter country name"
+          value={countryName}
+          onChange={(e) => dispatch(setCountryName(e.target.value))}
+        />
+      </div>
+      <p className="text-white bg-sky-800 p-1 font-lato">country by Category</p>
+      <ul className="grid grid-cols-2 md:grid-cols-3 lg:grid-cols-5">
+        {countryData.length === 0 ? (
+          <li>No country data available.</li>
+        ) : (
+          countryData.map((country) => (
+            <Link to={`/details/${country.alpha3Code}`} key={country.alpha3Code}>
+              <li className="relative odd:bg-sky-700 even:bg-sky-600 text-white h-64">
+                <img src={country.flags.png} alt={`${country.name} flag`} className="w-64 md:w-72" />
+                <div className="p-2 flex flex-col items-end">
+                  <strong className="font-bold text-2xl uppercase">
+                    {country.name.slice(0, 10)}
+                  </strong>
+                  <p className="flex gap-1">
```

Comment:
- You seem to be rendering your components filtered when your app initially renders, this is not the expected behavior of the application. By default, you set the value of the filter to `Am`, and each time I clear the filter, nothing is rendered, please fix this and ensure by default, even when the filter is empty, data is rendered on your application 🙏 
![dwded](https://github.com/axcell-0/react-metrics-capstone/assets/73607512/f9bde1be-54a5-42cb-ae8c-6e376c444e13)
- Also, please note that according to the project requirements, your filter should filter the data already gotten from the API and not fetch new data from the API. Please fix this 🙏